### PR TITLE
keep null in x,y so regl knows the end of polygon

### PR DESCRIPTION
**Pull request by ruijin**
fix for issue #2291

### CODE REVIEW

**Comment by archmoj** [comment]
Thanks very much for the PR! :1st_place_medal: 
Could you please add a mock and baseline to our image tests similar to this codepen?
https://github.com/plotly/plotly.js/issues/2291#issue-291306226

Here is more info https://github.com/plotly/plotly.js/blob/master/CONTRIBUTING.md#image-pixel-comparison-tests

**Comment by archmoj** [comment]
Unfortunately these changes do not appear to fix the problem on [this codepen](https://codepen.io/MojtabaSamimi/pen/YzGQaQa). 
@ruijin could you guide me to a case (data/layout) where your patch fixed the problem, please?

**Comment by ruijin** [comment]
I will try to provide a case. But just in case, did you test it together with the regl-line2d PR I submitted together? https://github.com/gl-vis/regl-line2d/pull/49

**Comment by ruijin** [comment]
I modify the existing gl2d_scatter_fill_self_next test to add test with >2 segment, which exposed the issue. Below is the screen show on my local run before and after the fix. You will need the PR in regl (https://github.com/gl-vis/regl-line2d/pull/49) to see it.

Before
plotly_pre_fix
After
plotly_post_fix

**Comment by archmoj** [comment]
> I will try to provide a case. But just in case, did you test it together with the regl-line2d PR I submitted together? [gl-vis/regl-line2d#49](https://github.com/gl-vis/regl-line2d/pull/49)

Please edit the link for `regl-line2d` in `package.json`
So instead of 
```
"regl-line2d": "^3.0.18",
```
you should provide the last commit hash and address to your fork of `regl-line2d` which is 
```
"regl-line2d": "git://github.com/ruijin/regl-line2d.git#5d544746bbbcf09223281a0da5549a485782ffe1",
```
Then run `npm install` so that `package-lock.json` is updated.
Finally please commit the changes to `package.json` and `package-lock.json` files and push.

**Comment by ruijin** [comment]
The CI kept failing at random place, it does seem like triggered by the code I touched?

**Comment by archmoj** [comment]
> The CI kept failing at random place, it does seem like triggered by the code I touched?

I rerun the tests. The jasmine ones now pass. But there is a problem catch by image test.
Could you investigate why gl2d_scatter_fill_self_next does not render?

**Comment by ruijin** [comment]
The issue of baseline is that I used Array.findIndex in the new code, and the image test server used by the test framework is a bit old and does not support that. So I use a polyfill instead. It should be fixed now.

File: package.json [hunk 1]
```diff
@@ -110,7 +110,7 @@
     "polybooljs": "^1.2.0",
     "regl": "^1.6.1",
     "regl-error2d": "^2.0.11",
-    "regl-line2d": "^3.0.18",
+    "regl-line2d": "git://github.com/ruijin/regl-line2d.git#d66c94274f53e7683f501f2192bde68c93e2221c",
```
Comment (archmoj) [inline package.json:113]: Please upgrade to 
```
"regl-line2d": "^3.1.0",
```

**Review by ruijin — COMMENTED** [review]
update regl-line2d to latest official version

**Comment by archmoj** [comment]
Nicely done.
:dancer:

**Pull request merged**